core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -0,0 +1,197 @@
+use crate::prelude::*;
+use models::AccountIdentifier;
+use models::{Operation, OperationIdentifier};
+use radix_engine_interface::blueprints::account::{
+    AccountTryDepositOrAbortManifestInput, AccountWithdrawManifestInput,
+};
+use radix_transactions::manifest::{CallMethod, TakeFromWorktop};
+use radix_transactions::validation::TransactionValidator;
+
+pub(crate) async fn handle_construction_parse(
+    state: State<MeshApiState>,
+    Json(request): Json<models::ConstructionParseRequest>,
+) -> Result<Json<models::ConstructionParseResponse>, ResponseError> {
+    assert_matching_network(&request.network_identifier, &state.network)?;
+
+    let transaction_bytes = hex::decode(&request.transaction).map_err(|_| {
+        ResponseError::from(ApiError::InvalidTransaction)
+            .with_details(format!("Invalid transaction hex: {}", &request.transaction))
+    })?;
+    let (instructions, signers) = if request.signed {
+        let transaction =
+            NotarizedTransactionV1::from_raw(&RawNotarizedTransaction::from_vec(transaction_bytes))
+                .map_err(|_| {
+                    ResponseError::from(ApiError::InvalidTransaction)
+                        .with_details(format!("Invalid transaction: {}", &request.transaction))
+                })?;
+        let validated = transaction
+            .prepare_and_validate(&TransactionValidator::new_with_latest_config(
+                &state.network,
+            ))
+            .map_err(|e| {
+                ResponseError::from(ApiError::InvalidTransaction)
+                    .with_details(format!("Invalid transaction: error = {:?}", e))
+            })?;
+
+        let instructions = transaction.signed_intent.intent.instructions.0;
+        let signers = validated.signer_keys;
+        (instructions, signers)
+    } else {
+        let prepared_intent = PreparedIntentV1::prepare(
+            &RawTransactionIntent::from_vec(transaction_bytes),
+            &PreparationSettings::latest(),
+        )
+        .unwrap();
+        #[allow(deprecated)]
+        let instructions = prepared_intent.instructions.inner.0;
+        let signers = index_set_new();
+        (instructions, signers)
+    };
+
+    let database = state.state_manager.database.snapshot();
+    let operations = parse_instructions(
+        &instructions,
+        &MappingContext::new(&state.network),
+        database.deref(),
+    )?;
+
+    // See https://docs.cdp.coinbase.com/mesh/docs/models#constructionparseresponse for field
+    // definitions
+    Ok(Json(models::ConstructionParseResponse {
+        operations,
+        signers: None,
+        account_identifier_signers: Some(
+            signers
+                .into_iter()
+                .map(|x| AccountIdentifier {
```

Comment:
Ditto - should use a conversion function here to create this; for consistency.

Reply:
Done